include/swift/Parse/Parser.h

```diff
@@ -2075,6 +2039,14 @@ class Parser {
   bool parseLegacyTildeCopyable(SourceLoc *parseTildeCopyable,
                                 ParserStatus &Status,
                                 SourceLoc &TildeCopyableLoc);
+
+  //===--------------------------------------------------------------------===//
+  // ASTGen support.
+
+  ParserResult<TypeRepr> parseTypeReprFromSyntaxTree();
```

Comment:
Just reading the function declaration, I’m thinking: Which syntax tree is this parsing from? I think a doc comment could help explain this (or a need a better name).